compiler/rustc_hir_analysis/src/astconv/errors.rs

```diff
@@ -197,7 +201,34 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             }
         }
 
-        err.span_label(span, format!("associated type `{assoc_name}` not found"));
+        // If we still couldn't find any associated type, just list them all.
+
+        if all_candidate_names.is_empty() {
+            err.help(format!(
+                "`{ty_param_name}` has no associated type, try removing `{assoc_name}`"
+            ));
+            return err.emit();
+        }
+
+        let msg = if all_candidate_names.len() > 1 {
```

Comment:
I don't think we should suggest anything if there's >1 candidate.

Reply:
```rs
trait Qux {
    type A;
    type B;
}

fn quux<T>() where T: Qux<A = u32, ZZ = u32> {}
```
gives
```
error[E0220]: associated type `ZZ` not found for `Qux`
  --> E0220.rs:14:36
   |
14 | fn quux<T>() where T: Qux<A = u32, ZZ = u32> {}
   |                                    ^^
   |
help: `Qux` has the following associated types
   |
14 | fn quux<T>() where T: Qux<A = u32, A = u32> {}
   |                                    ~
14 | fn quux<T>() where T: Qux<A = u32, B = u32> {}
   |                                    ~
```
I think this is still helpful?